### PR TITLE
adding and implementing logo component

### PR DESCRIPTION
**Pull request by jsaalfeld**
There is a `TODO` left, because the path to the default variables is yet still empty when loading the page. So the default must be set via code.

### CODE REVIEW

File: client/src/app/shared/components/logo/logo.component.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    /**
+     * decides based on the actionString how to display the logo
+     */
+    @Input()
+    public inputAction: string;
+
+    /**
+     * The consotructor
+     * @param mmservice The Media Manage Service
```
Comment (tsiegleauq) [inline client/src/app/shared/components/logo/logo.component.ts]: line spacing

File: client/src/app/shared/components/logo/logo.component.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        let path = '';
+        if (this.mmservice !== undefined) {
+            if (MediaManageService.isImageConfigObject(this.mmservice.getMediaConfig(this.inputAction))) {
+                const imageConfig = this.mmservice.getMediaConfig(this.inputAction);
+                path = imageConfig.path;
+            }
+        }
+        if (path === '') {
+            // TODO: Replace with config.default
+            path = '/assets/img/openslides-logo-h.svg';
```
Comment (tsiegleauq) [inline client/src/app/shared/components/logo/logo.component.ts]: @FinnStutzenstein 
do you know if the default logo is written in a config or a constant?

File: client/src/app/site/mediafiles/services/media-manage.service.ts [hunk 3]
```diff
@@ -106,4 +114,12 @@ export class MediaManageService {
     public getMediaConfig(action: string): ImageConfigObject | FontConfigObject {
         return this.config.instant(action);
     }
+
```
Comment (tsiegleauq) [inline client/src/app/site/mediafiles/services/media-manage.service.ts]: remove

File: client/src/app/site/mediafiles/services/media-manage.service.ts [hunk 4]
```diff
@@ -41,6 +41,14 @@ interface ManagementPayload {
     providedIn: 'root'
 })
 export class MediaManageService {
+    /**
+     * Checks if an image is an imageConfig Object
+     */
+    public static isImageConfigObject(object: any): object is ImageConfigObject {
```
Comment (tsiegleauq) [inline client/src/app/site/mediafiles/services/media-manage.service.ts]: dont use static

File: client/src/app/site/site.component.scss [hunk 5]
```diff
@@ -36,12 +36,14 @@ mat-sidenav-container {
 .os-logo-container {
     width: 200px;
     margin-left: 10px;
-    background-image: url(/assets/img/openslides-logo-h.svg);
-    background-size: contain;
-    background-repeat: no-repeat;
-    background-position: center;
+    // background-image: url(/assets/img/openslides-logo-h.svg);
```
Comment (tsiegleauq) [inline client/src/app/site/site.component.scss]: remove

File: client/src/app/site/mediafiles/services/media-manage.service.ts [hunk 6]
```diff
@@ -79,6 +79,15 @@ export class MediaManageService {
         return this.httpService.put<void>(restPath, payload);
     }
 
+    /**
+     * Checks if an image is an imageConfig Object
+     */
+    public isImageConfigObject(object: any): object is ImageConfigObject {
```
Comment (tsiegleauq) [inline client/src/app/site/mediafiles/services/media-manage.service.ts]: @returns

**Comment by jsaalfeld** [comment]
>    * [x] If there is no image at all, there will an empty image at the "footer"
> 
>     * [ ] please remove the blue line that appears when you click at the OpenSlides logo.
> 
>     * [x] logos which fill their full vertical and horizontal space are too tight to the upper cornor of the screen
> 
>     * [x] Logos do not keep their original aspect ratio

The blue line is a problem with the child-container ignoring the maximal height. It works for images, for now, with an aspect ratio of `> 1.5625`.

Any help with the CSS magic of images with smaller ratios is appreciated.

The `site.component.scss` with the `.os-footer-logo-container` and `.os-logo-container` cause the issues. It takes the maximal width of `250px` but ignores the maximal height of the images with `160px`. This means for Images with a ratio of 1, that the displayed image will be `250px` by `250px`, but the underlaying container has the maximum size of `250px` by `160px`.

**Comment by jsaalfeld** [comment]
I've seen to fix the original issue by adding `display: flex` to the logo. But the logo-container itself is not aligned vertically and the image in the container is not aligned horizontally. But the aspect ratio and size are fine now:

https://stackblitz.com/edit/angular-ktuur2

**Comment by jsaalfeld** [comment]
With just about *6h* of developer blood sweat and tears from 4 different devs, we managed to center the header logo. What a fight.

Therefore:

> 
>    * [x] If there is no image at all, there will an empty image at the "footer"
> 
>    * [x] please remove the blue line that appears when you click at the OpenSlides logo.
> 
>    * [x] logos which fill their full vertical and horizontal space are too tight to the upper cornor of the screen
> 
>    * [x] Logos do not keep their original aspect ratio

**Pull request merged**